src/rust/engine/engine_pyo3/src/externs/interface/fs.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::path::PathBuf;
+
+use fs::{GlobExpansionConjunction, PathGlobs, PreparedPathGlobs, StrictGlobMatching};
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+pub(crate) fn register(m: &PyModule) -> PyResult<()> {
+  m.add_function(wrap_pyfunction!(match_path_globs, m)?)?;
+  Ok(())
+}
+
+struct PyPathGlobs(PathGlobs);
+
+impl PyPathGlobs {
+  fn parse(self) -> PyResult<PreparedPathGlobs> {
+    self.0.clone().parse().map_err(|e| {
```

Comment:
Oh! This `clone()` might explain the slowdown in the contrived benchmark. We do not have that `clone()` in Rust-CPython version and clone an `Arc` instead.

Reply:
Nope, that's not it. And the `PathGlobs` is cheap in the example. It's the `paths: Vec<String>` that's so huge and would be painful to clone.

Reply:
Because this `fn` takes `self`, it shouldn't need to clone: it can consume `self` (including `self.0`).

Reply:
> presumably the glob which failed to parse? You might not need to clone the rest of the globs.

Yeah, although we want to show what `self` was in the error message so it's more useful. Not technically necessary, but I think it's fine because the `PathGlobs` object is usually pretty small (a vec of a few strings).

@stuhood thoughts?